gef-ui/server/controllers/exporters-address/index.test.js

```diff
@@ -175,20 +184,62 @@ describe('Validate Exporters Address', () => {
     const mockRejection = { status: 422, data: [{ errMsg: 'Message', errRef: 'Reference' }] };
 
     mockRequest.body.correspondence = 'true';
-    mockRequest.body.postcode = 'FFFFF';
+    mockRequest.body.postcode = postcode;
 
     api.getApplication = () => Promise.resolve(mockApplicationResponse);
     api.getExporter = () => Promise.resolve(mockExporterResponse);
     api.getAddressesByPostcode = () => Promise.resolve(mockRejection);
     await validateExportersAddress(mockRequest, mockResponse);
     expect(mockResponse.render).toHaveBeenCalledWith('partials/exporters-address.njk', expect.objectContaining({
-      companyName: undefined,
+      companyName,
       errors: expect.any(Object),
       correspondence: 'true',
-      registeredAddress: undefined,
+      registeredAddress,
     }));
   });
 
+  it('shows user an error, allowing retry, if there is an issue with the address api', async () => {
+    const mockRequest = new MockRequest();
+    const mockResponse = new MockResponse();
+    const mockApplicationResponse = new MockApplicationResponse();
+    const mockExporterResponse = new MockExporterResponse();
+    const mockedRejection = { response: { status: 500, message: 'Whoops' } };
+    const expectedMsg = 'Error looking up postcode. Try again.';
+    const correspondence = 'true';
+
+    mockRequest.body.correspondence = correspondence;
+    mockRequest.body.postcode = postcode;
+    mockExporterResponse.details.companyName = companyName;
+
+    api.getApplication = () => Promise.resolve(mockApplicationResponse);
+    api.getExporter = () => Promise.resolve(mockExporterResponse);
+    api.getAddressesByPostcode = () => Promise.reject(mockedRejection);
+
+    await validateExportersAddress(mockRequest, mockResponse);
+    expect(mockResponse.render).toHaveBeenCalledWith(
+      'partials/exporters-address.njk',
+      {
+        errors: {
+          errorSummary: [
```

Comment:
I'd consider mocking the response of this personally but not important

Reply:
fair point, but for now I think its ok to just have a static object to test against

Reply:
yeah happy with that - just a minor thing really. Can just keep some of these objects simpler